### PR TITLE
Training with disabilities view on courses

### PR DESCRIPTION
**Pull request by CatalinVoineag**
### Context

The course show page on find and the preview page on publish contains
too much information that might be off-putting for the user.

This commit tries to fix this by putting this section into its own view
accessible from the course page

### Changes proposed in this pull request

347fce874492a8cb967521fa7b5679f0128af0cd This implements the view/controller/routes changes

91f3fa13a3c6efc513817597003e403847f3179a This is a bit of a refactor of the `goto` methods that tell the `about_your_organisation` form where to redirect

Description of 91f3fa13a3c6efc513817597003e403847f3179a:
There are a few `goto` methods that tell the about your organisation
form where to redirect the user after update or when the user clicks the
back links.

Basically, both back links and update success methods can go to the same
path based on the `goto` param.

Because of this and the ever growing of these `goto` params I moved some
logic that calculates the redirection in the
about_your_organisation_form. This way it can be tested properly.

### Guidance to review

Review the code and check if the training for disabilities page works

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased main
- [x] Cleaned commit history
- [x] Tested by running locally
- [ ] Inform data insights team due to database changes

https://github.com/DFE-Digital/publish-teacher-training/assets/11318084/cc95a55b-3e85-4e7c-9c7f-213f831a3fdf

### CODE REVIEW

File: app/controllers/concerns/goto_preview.rb [hunk 1]
```diff
@@ -8,5 +8,4 @@ def param_form_key
   end
 
   def goto_preview? = params.dig(param_form_key, :goto_preview) == 'true'
-  def goto_provider? = params.dig(param_form_key, :goto_provider) == 'true'
```
Comment (CatalinVoineag) [inline app/controllers/concerns/goto_preview.rb:11]: This is no longer needed after the refactor 91f3fa13a3c6efc513817597003e403847f3179a

File: app/helpers/find/goto_preview_helper.rb [hunk 2]
```diff
@@ -14,8 +14,8 @@ def goto_provider_value(param_form_key:, params:)
       params[:goto_provider] || params.dig(param_form_key, :goto_provider)
     end
 
-    def goto_provider?(param_form_key:, params:)
```
Comment (CatalinVoineag) [inline app/helpers/find/goto_preview_helper.rb:17]: No longer needed

**Comment by CatalinVoineag** [comment]
[Simplify course pages 5/6 - Move "Training with disabilities and other needs" to a separate page](https://trello.com/c/ip4O0VLc/1796-simplify-course-pages-5-6-move-training-with-disabilities-and-other-needs-to-a-separate-page)

File: app/controllers/publish/providers_controller.rb [hunk 3]
```diff
@@ -54,29 +54,26 @@ def details
...
-              params[:course_code] || params.dig(param_form_key, :course_code)
-            )
-          )
-        elsif goto_preview?
-          redirect_to preview_publish_provider_recruitment_cycle_course_path(provider.provider_code, provider.recruitment_cycle_year, (params[:course_code] || params.dig(param_form_key, :course_code)))
-        else
-          flash[:success] = I18n.t('success.published')
-          redirect_to(details_publish_provider_recruitment_cycle_path(provider.provider_code, provider.recruitment_cycle_year))
-        end
+        redirect_to @about_form.update_success_path
```
Comment (elceebee) [inline app/controllers/publish/providers_controller.rb:75]: This is so much cleaner, nice one!

File: app/views/publish/providers/about.html.erb [hunk 4]
```diff
@@ -10,19 +10,7 @@
...
-              @provider.recruitment_cycle_year,
-              params[:course_code] || params.dig(f.object_name.to_sym, :course_code)
-            )
-          ) %>
-       <% elsif goto_preview?(param_form_key: f.object_name.to_sym, params:) %>
-        <%= govuk_back_link_to(preview_publish_provider_recruitment_cycle_course_path(@provider.provider_code, @provider.recruitment_cycle_year, (params[:course_code] || params.dig(f.object_name.to_sym, :course_code)))) %>
-        <% else %>
-          <%= govuk_back_link_to(details_publish_provider_recruitment_cycle_path(@provider.provider_code, @provider.recruitment_cycle_year)) %>
-        <% end %>
+        <%= govuk_back_link_to(@about_form.back_path) %>
```
Comment (elceebee) [inline app/views/publish/providers/about.html.erb:13]: ✨

**Review by elceebee — APPROVED** [review]
Great work taking the time to refactor this goto behaviour. I've tested the back links and other functionality in the review app and it all looks good to me. 👍

**Comment by kelliedesigner** [comment]
### Content change to new view

- Update title of the new view to `Training with disabilities and other needs at [PROVIDER NAME]`

| Example of new view page title |
| --- |
| Screenshot 2024-07-11 at 11 41 43 |

### Content changes to course page (not detailed in the original ticket)

- Update the contents list to `Training with disabilities`

| Contents list example |
| --- |
| Screenshot 2024-07-11 at 11 36 40 |

- Update the `<h2>` to `Training with disabilities`
- Add a full stop after the sentence, outside of the `<a>`

| Example of training with disabilities section |
| --- |
| Screenshot 2024-07-11 at 11 37 57 |

**Comment by kelliedesigner** [comment]
Looks good to me!

**Review by kelliedesigner — APPROVED** [review]
